pkg/gateway/model_build_targetgroup.go

```diff
@@ -196,10 +197,18 @@ func (t *targetGroupModelBuildTask) BuildTargetGroup(ctx context.Context) error
 		},
 	}
 
-	glog.V(6).Infof("Found backend service port %v , intval %v \n", svc.Spec.Ports[0].TargetPort,
-		svc.Spec.Ports[0].TargetPort.IntVal)
-
-	tgSpec.Config.Port = svc.Spec.Ports[0].TargetPort.IntVal
+	var backendServicePort int32
+	if port := svc.Spec.Ports[0].TargetPort.IntVal; port != 0 {
+		backendServicePort = port
+		glog.V(6).Infof("Found backend service port %v, intval %v\n", svc.Spec.Ports[0].TargetPort,
+			svc.Spec.Ports[0].TargetPort.IntVal)
+	} else {
+		// Fallback to default HTTP port. This can be any number, because we always use the port defined in targets.
```

Comment:
if port is irrelevant for target group lattice config, can we just always use the "else" portion of logic?